lib/block/blockupload.js

```diff
@@ -128,15 +128,11 @@ class BlockUpload extends EventEmitter {
      */
     async uploadFiles(options = {}) {
         const preferredPartSize = options.preferredPartSize || DEFAULT_PART_SIZE;
-        const maxConcurrent = (options.concurrent && options.maxConcurrent) || DEFAULT_MAX_CONCURRENCY;
-        const retryOptions = {
-            retryMaxCount: options.retryCount || DEFAULT_RETRY_COUNT,
-            timeout: options.timeout,
-            retryAllErrors: options.retryAllErrors,
-            retryMaxDuration: options.retryMaxDuration,
-            retryInitialDelay: options.retryInitialDelay,
-            retryBackoff: options.retryBackoff,
-        };
+        const maxConcurrent = options.maxConcurrent || DEFAULT_MAX_CONCURRENCY;
+
+        if (!options.retryMaxCount) {
+            options.retryMaxCount = DEFAULT_RETRY_COUNT;
+        }
```

Comment:
instead of filtering everything here, instead I filtered it outside of this function in the exposed api `uploadFileConcurrently`